--- old/src/solaris/classes/sun/awt/X11/XWindowPeer.java	2013-04-22 17:10:34.379178316 +0300
+++ new/src/solaris/classes/sun/awt/X11/XWindowPeer.java	2013-04-22 17:10:33.942514015 +0300
@@ -1887,7 +1887,9 @@
         switch (getWindowType())
         {
             case NORMAL:
-                typeAtom = protocol.XA_NET_WM_WINDOW_TYPE_NORMAL;
+                typeAtom = (ownerPeer == null) ? 
+                               protocol.XA_NET_WM_WINDOW_TYPE_NORMAL : 
+                               protocol.XA_NET_WM_WINDOW_TYPE_DIALOG;
                 break;
             case UTILITY:
                 typeAtom = protocol.XA_NET_WM_WINDOW_TYPE_UTILITY;
